Ica example 2 tutorial

neurogima opened the pull request:
#### Reference issue
Fixes #9003


#### What does this implement/fix?
combine ICA example into the ICA tutorial


#### Additional information
Relevant information to combine the code:
- reuse the filt_raw object instead of importing again the raw fif
- change the number of components in accordance to ICA on raw
- kept the band pass filter like in the example

neurogima commented:
Apologies,I forgot to run flake8 on the tutorial file and I left two lines more between the last code block and the reference block. The make of the HTML I ran locally completed successfully though. What is the procedure in this cases?

agramfort commented:
you need to continue pushing to your branch and it will update the PR.

once the doc has been successfully built you will see a link to the
"artifact". Click
on it and share with us the URL of the rendered page to ease our review.

thx

>

sappelhoff commented on tutorials/preprocessing/plot_40_artifact_correction_ica.py:
```diff
@@ -585,7 +585,59 @@
+# Plot the estimated source of detected ECG related components:
+ica.plot_sources(filt_raw, picks=ecg_inds)
+
+
+
```
```suggestion
```

neurogima commented:
this is the rendered version:
https://26206-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/preprocessing/plot_40_artifact_correction_ica.html?highlight=repairing%20artifacts

drammock commented:
I ran out of time to do a careful review today but in principle it looks good, mostly just some small changes needed to smooth out the wording.  I'll try to look again tomorrow morning if nobody else beats me to it.

britta-wstnr commented on tutorials/preprocessing/plot_40_artifact_correction_ica.py:
```diff
@@ -585,7 +585,57 @@
+# Example: Compute ICA components on epochs
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# ICA is now fit to MEG epochs data instead of the raw data.
```
```suggestion
# ICA is now fit to epoched MEG data instead of the raw data.
```

britta-wstnr commented on tutorials/preprocessing/plot_40_artifact_correction_ica.py:
```diff
@@ -585,7 +585,57 @@
+
+# peak-to-peak amplitude rejection parameters
+reject = dict(grad=4000e-13, mag=4e-12)
+# longer + more epochs for more artifact exposure
```
```suggestion
# create longer and more epochs for more artifact exposure
```

What do you mean with "more"? Is there an example above where less epochs get selected? (I tried to look but might have missed it)

britta-wstnr commented on tutorials/preprocessing/plot_40_artifact_correction_ica.py:
```diff
@@ -585,7 +585,57 @@
+ica.plot_components(ecg_inds)
+
+###############################################################################
+# Plot properties of ECG components:
```
```suggestion
# Plot the properties of the ECG components:
```

britta-wstnr commented on tutorials/preprocessing/plot_40_artifact_correction_ica.py:
```diff
@@ -585,7 +585,57 @@
+ica.plot_properties(epochs, picks=ecg_inds)
+
+###############################################################################
+# Plot the estimated source of detected ECG related components:
```
```suggestion
# Plot the estimated sources of detected ECG related components:
```

britta-wstnr commented on tutorials/preprocessing/plot_40_artifact_correction_ica.py:
```diff
@@ -585,7 +585,57 @@
 # References
-# ----------
+# ^^^^^^^^^^
+#
```
Fine to delete that blank line here.

britta-wstnr commented on tutorials/preprocessing/plot_40_artifact_correction_ica.py:
```diff
@@ -585,7 +585,57 @@
 # .. _`this EEGLAB tutorial`: https://labeling.ucsd.edu/tutorial/labels
 
+
+###############################################################################
+# Example: Compute ICA components on epochs
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```suggestion
###############################################################################
# Compute ICA components on epochs
```

britta-wstnr replied:
@drammock or was the intention to have an example within a tutorial?

britta-wstnr replied:
Okay, let's just make a decision: drop the "Example".

britta-wstnr commented on tutorials/preprocessing/plot_40_artifact_correction_ica.py:
```diff
@@ -585,7 +585,57 @@
+# We assume that the non-stationary EOG artifacts have already been removed.
+# The sources matching the ECG are automatically found and displayed.
+#
+# .. note::
+#     This example does quite a bit of processing, so even on a
+#     fast machine it can take about a minute to complete.
```
I think this could be rephrased but I do not have a good alternative. @drammock how is this usually communicated in tutorials or examples?

britta-wstnr reviewed (commented):
Did a pass on the wording of the comments, some small suggestions there.

Rendering looks good:
https://26206-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/preprocessing/plot_40_artifact_correction_ica.html

britta-wstnr commented:
@neurogima in the meanwhile, the ICA example's `max_iter='auto'` got fixed on `main` and you have a merge conflict that you'll have to resolve - that is why CIs fail now.

hoechenberger commented:
Is this ready for review / good to merge?

neurogima commented:
@hoechenberger just two very minor things are waiting for your review related to britta's last two comments: including or not the world example and the note about the computing time

britta-wstnr commented on tutorials/preprocessing/plot_40_artifact_correction_ica.py:
```diff
@@ -585,7 +585,56 @@
+# We assume that the non-stationary EOG artifacts have already been removed.
+# The sources matching the ECG are automatically found and displayed.
+#
+# .. note::
+#     This example does quite a bit of processing, so even on a
+#     fast machine it can take about a minute to complete.
```
```suggestion
# .. note::
#     This example is computationally intensive, so it might take a few minutes
#     to complete.
```

britta-wstnr reviewed (commented):
Made a decision on the last open questions.
Test failure looks unrelated to me, maybe fixed in `main`, @hoechenberger ?

drammock commented:
> Test failure looks unrelated to me, maybe fixed in `main`

prior test failures were indeed ones already fixed in main, they should hopefully disappear on this latest commit.

drammock commented:
one other request @neurogima is that we should *remove* the old example file since it is now redundant with this tutorial.  `git rm examples/preprocessing/plot_run_ica.py` should do it.

neurogima commented:
> one other request @neurogima is that we should _remove_ the old example file since it is now redundant with this tutorial. `git rm examples/preprocessing/plot_run_ica.py` should do it.

this is the rendered version:
https://26520-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/preprocessing/plot_40_artifact_correction_ica.html?highlight=repair%20artifact

it was built after I've pulled, removed the file, made a last style modification on the title and committed, rebased on main and pushed.

drammock commented:
awesome, thanks @neurogima!
